modules/team.js

```diff
@@ -58,6 +63,19 @@ export const joinTeam = (teamId, name) => async (dispatch, getState) => {
   }
 };
 
+export const fetchTeam = (id) => async (dispatch) => {
+  try {
+    const res = await API().get(`teams/${id}`);
+    dispatch({
+      type: SET_TEAM,
+      payload: res.data,
+    });
+  }
+  catch (err) {
+    toast.error(errorToString(err.response.data.error));

```

Comment:
Je vous conseil de créer un handler pour les requêtes à l'API, genre un handleError(err) qui 1 affiche l'erreur dans la console quand env!=production et 2 affiche le message d'erreur correspondant.
Le problème ici c'est que dans le cas rare où l'api tombe, err.response est undefined, donc err.response.data => cant get property data of undefined. Si tu gères ça dans un handleError tu peux faire un truc chiant genre if(err && err.response && err.response.data && etc)

En plus une fois que c'est fait tu l'appels partout et tu réfléchis plus

Reply:
Yep bien vu! clairement c'est chiant de faire toast.error à chaque try catch